### PR TITLE
Suppress logging of expected authentication pipeline-related exceptions

### PR DESCRIPTION
**Pull request by acrellin**
Depends on https://github.com/cesium-ml/baselayer/pull/202 

Closes https://github.com/skyportal/skyportal/pull/1744

### CODE REVIEW

File: skyportal/app_server.py [hunk 1]
```diff
@@ -86,6 +87,35 @@
 
 log = make_log('app_server')
 
+
+class CustomApplication(tornado.web.Application):
+    def log_request(self, handler):
+        """Writes a completed HTTP request to the logs.
+        By default writes to the python root logger.  To change
```
Comment (stefanv) [inline skyportal/app_server.py:94]: What is "the python root logger"?

File: skyportal/app_server.py [hunk 2]
```diff
@@ -86,6 +87,35 @@
 
 log = make_log('app_server')
 
+
+class CustomApplication(tornado.web.Application):
+    def log_request(self, handler):
+        """Writes a completed HTTP request to the logs.
```
Comment (stefanv) [inline skyportal/app_server.py:93]:
```suggestion
        """Writes a completed HTTP request to the logs.
        
```

File: skyportal/app_server.py [hunk 3]
```diff
@@ -86,6 +87,35 @@
...
+        By default writes to the python root logger.  To change
+        this behavior either subclass Application and override this method,
+        or pass a function in the application settings dictionary as
+        ``log_function``.
+        """
+        if "google-oauth2" in str(handler._request_summary()):
```
Comment (stefanv) [inline skyportal/app_server.py:99]: I'm not confident about this method; what code is this based on?  It feels a bit magical.  What does this do that we need, that is not currently being done?

File: skyportal/app_server.py (hunk 1 above)
Comment (acrellin) [reply skyportal/app_server.py:94]: this is just copy-pasted from the tornado source code, with minor changes

File: skyportal/app_server.py (hunk 3 above)
Comment (acrellin) [reply skyportal/app_server.py:99]: @stefanv We've discussed this. You said you don't want to see the "access error 500" messages in the logs when they're associated with the PSA pipeline -- this filters those out (the URL always contains "google-oauth2" in those cases). What's magical about it? I'm happy for you to propose an alternate implementation.

Comment (stefanv) [reply skyportal/app_server.py:92]: Looks good now!  Please also add a comment to explain why we do this.

**Review by stefanv — APPROVED** [review]
Thanks, Ari!

**Comment by acrellin** [comment]
@stefanv okay to merge this one?

**Comment by acrellin** [comment]
@stefanv pinging you again here

File: skyportal/app_server.py (hunk 3 above)
Comment (stefanv) [reply skyportal/app_server.py:96]: The comment says "message text" and the comment above says "in the request route".  Request route makes the most sense to me; is `_request_summary()` (private method?) the best way to get to that?

Comment (acrellin) [reply skyportal/app_server.py:96]: Comment updated. Yes, why is `_request_summary` not a good way to get that?

**Comment by stefanv** [comment]
It's a private method; was wondering if that's the best way to get the route?

**Comment by acrellin** [comment]
@stefanv okay, now uses a non-private attribute

**Comment by stefanv** [comment]
Beautiful!

**Pull request merged**